lib/mongoose-core.module.ts

```diff
@@ -95,10 +95,12 @@ export class MongooseCoreModule implements OnApplicationShutdown {
 
         return await defer(async () =>
           mongooseConnectionFactory(
-            mongoose.createConnection(
-              mongooseModuleOptions.uri,
-              mongooseOptions as any,
-            ),
+            mongooseModuleOptions.uri
```

Comment:
To avoid introducing breaking changes, can we please just remove this condition and pass `mongooseModuleOptions.uri` even if it's `undefined`?

Reply:
How is this condition a breaking change?

Reply:
Previously, we passed `mongooseModuleOptions.uri `regardless of whether it's truthy or not (e.g. `undefined`). Now, you are calling `mongoose.createConnection()` in this case

Reply:
Yeah, saw that. I can make the change. Absolutely no problem :)

However I'm generally interested to know why that is a breaking change?

Because I thought mongoose will just check whether the `uri` is `falsy` in their code and then use the default URI or not.

I actually dug into the mongoose code to verify this but was unable to find it. So I'm asking you because I think you know more of the matter :)

Reply:
@BorntraegerMarc oh, so I'm actually referring to this code and nothing else. Perhaps, mongoose is using a fallback to the default value anyway, but I'm just trying to avoid breaking changes on our side :)

Reply:
Gotcha. I made the change now :)

If you approve I will squash & merge with the PR so we have a clean commit history on master.

Let me know what you think about my idea here: https://github.com/nestjs/mongoose/issues/226#issuecomment-576004329

I think this is a scenario that could be handled better if we align optional / mandatory options with mongoose. Though it will require a breaking change.